### ISSUE TITLE
Install - try out hassles, cannot get to work at all.

### ISSUE THREAD

**Issue by neocogent** [chris-belcher/electrum-personal-server#67]
This is meant as feedback after trying to use for the first time.

I git cloned the repo to my local bitcoin node. I tried following the steps described in the readme. However, there is no ./server.py command so this does not work. After digging thru issues and looking around I found that things have changed but the readme is not updated. You have to use pip install or setup.py install now and a different server command. pip install did not work for me. It does not install correctly the ...1.4-dev something or other. But setup.py install did work. 

I had configured the config.cfg file to use my existing rpcuser, rpcpassword values and my ypub from my wallet. When I finally run electrum-personal-server command I just the msg:

```
INFO:2018-11-07 09:13:15,431: Starting Electrum Personal Server
INFO:2018-11-07 09:13:15,431: Logging to /tmp/electrumpersonalserver.log
WARNING:2018-11-07 09:13:15,436: Unable to find .cookie file, try setting `datadir` config
```
But the server was not running. I edited the config to have datadir but got same message as above saying to try the datadir config. No difference.

I thought maybe this is because not using rpcauth or cookie auth. I tried to change my bitcoin config but cannot find rpcuser.py command. Apparently it is not installed when bitcoin is installed from Ubuntu repo (though I'm still unsure). Anyway, seems I cannot generate an rpcauth line for config, so stuck with rpcuser/pwd config for now. Is this no longer supported by electrum-personal-server?

So my first try at using this is aborted. No running server. Not really sure why because no usable log message. Checked port with netstat and none being listened on, so not just me missing server in htop. 

Not sure what to do now except let you know it doesn't work for me.

**Comment by chris-belcher** [comment]
Thanks for the feedback. I'll try help you out.

Looks like you're stuck on rpc authentication.

To have it work with .cookie, you need to set datadir which on ubuntu by default would be `/home/username/.bitcoin/`. Alternatively you can set `rpcuser` and `rpcpass`, you have to set the same username and password in the `bitcoin.conf` file. Note that this is different to using rpcauth, I don't think EPS can even work with rpcauth, it uses either rpcuser/rpcpass or the .cookie file.

There may be helpful error messages in the debug log file, it's path is printed out by the server near the beginning.

**Comment by neocogent** [comment]
Thx. My system is using rpcuser/rpcpassword currently. I use bitcoin-cli to query info so that is working fine. 
My data dir is actually /NASZ/bitcoin (because I have it on a large ZFS pool). When I set that value EPS doesn't seem to use it as it gives the same msg whether set or not. I only tried rpcauth because what I already had wasn't working. I looked at the /tmp/electrumpersonalserver.log file and it only contains what was already dumped on console. No more useful details.

For the time being I deleted all EPS files (best I could) as I though maybe I'll try again later from scratch. Perhaps since I cloned the repo and started that way it doesn't get setup correctly. I do have experience developing in python and even releasing pypi packages but rather than try to dig deep and find out what's gone wrong I just went the lazy way and posted here.

Maybe it's misbehaving when both no .cookie and non-default data dir?

**Comment by chris-belcher** [comment]
If you use rpcuser/rpcpassword then you can't use .cookie and datadir. So try just setting rpcuser/pass.

**Comment by neocogent** [comment]
That was what I did from the start of this. It's only when it didn't work that I went looking for second options. According to docs it should have worked with my first try. Maybe I'll try again shortly with a fresh install not using git clone but from a release package. It's completely baffling to me and also generally when a python script fails you get some dump on the console but nothing here. When I have time I'll try again and try to debug what it's doing.

**Comment by andcoisqu** [comment]
@chris-belcher I'm not sure why you don't think EPS can't work with `rpcauth`. It will work the same way if you set `rpcuser` and `rpcpassword`, except the password is no longer in plaintext in the `bitcoin.conf` file. You will have to put it in plain text in the `config.cfg` file, however.

**Comment by neocogent** [comment]
I tried this again today (due to DDos attack - thought maybe I'll give it another go). Seems to be going further now as have ...[40%]... etc. being shown on console as it imports 2000 addresses (way more than I'd likely ever use). Maybe it'll complete then.

Just wondering if there is a cfg line to tell it where to log instead of /tmp? I guess I could make a symlink but probably temporary due to being /tmp dir.

**Comment by neocogent** [comment]
Oh. Except it does not update the history for multiple wallets. I have set multiple MPKs and rescanned. It show 6000 addresses being monitored. They seem to be all in the log. But also lots of messages about "address not found - means address not being monitored" etc.

The last section shows first 3 addresses of all 3 wallets I have config'd. They are correct. But still two of three wallets never update status and info. All dates and txs are very recent an I chose a date for rescan that was older by months. Not sure what is going on. Hmmm. I guess this still isn't working the way I'd like.
